content/chapters/compute/lab/support/sum-array-bugs/seg-fault/sum_array_processes.c

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <sys/wait.h>
+
+#include "include/generate_random_array.h"
+#include "include/array_utils.h"
+#include "utils/utils.h"
+
+static void calculate_array_part_sum(int *array, size_t start, size_t end,
+				     long *results, size_t tid)
+{
+	long sum_array = 0;
+	int *seg_fault_ptr = NULL;
+
+	for (size_t i = start; i < end; ++i)
+		sum_array += array[i];
+
+	if (tid == 2)
+		*seg_fault_ptr = 42;
+
+	results[tid] = sum_array;
+}
+
+static inline void *create_shared_results_array(size_t num_processes)
+{
+	/*
+	 * This buffer will be shared by the process that creates it and all its
+	 * children due to the `MAP_SHARED` argument
+	 */
+	return mmap(NULL, num_processes * sizeof(long),
+		    PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANON, -1, 0);
+}
+
+int main(int argc, char *argv[])
+{
+	int num_processes = atoi(argv[1]);
```

Comment:
The code seg faults here if run without arguments. Move the assignment below the `if` statement below.